packages/skill-api/src/server/default-payment-options.ts

```diff
@@ -0,0 +1,53 @@
+import Stripe from 'stripe'
+
+type StripeConfig = {
+  stripeSecretKey: string
+  apiVersion: '2020-08-27'
+}
+
+type StripeProvider = {name: 'stripe'; paymentClient: Stripe}
+type StripeProviderFunction = (options: StripeConfig) => StripeProvider
+
+type Paypal = 'paypal-client'
+type PaypalProvider = {name: 'paypal'; paymentClient: Paypal}
+type PaypalProviderFunction = (options: {
+  paypalSecretKey: string
+}) => PaypalProvider
+
+type PaymentProviderOptions = StripeProvider | PaypalProvider
+
+export type PaymentOptions = {
+  providers: {
+    stripe?: StripeProvider
+    paypal?: PaypalProvider
+  }
+}
```

Comment:
this file is mostly exploratory "writing on a whiteboard" -- `PaypalProvider` will probably need to be more appropriately named `LemonSqueezyProvider` for example.

I'm not married to any of these types, just trying to get the big picture sketched out in a way where the code also works.

Reply:
related in a few ways, but i like this example of creating types to describe adapters https://github.com/treehousedev/treehouse/blob/main/lib/backend/mod.ts